### PR TITLE
Deprecate pkg resources and fix mariadb healthcheck

### LINKED ISSUE #58: pkg_resources is deprecated as an API
You get ths message when running tests

### LINKED ISSUE #67: unhealthy mariadb status prevents running the dockerised demo
Replicate by following these instructions:

- Go to repo folder
- Run `make build`
- Run `make setup`

image

### PR DESCRIPTION
**Pull request by northwestwitch**
### Fixed
- Fix #58
- Fix #67 
 
## Review
- [x] Tests executed by CR
- [x] "Merge and deploy" approved by DN

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: chanjo_report/__init__.py [hunk 1]
```diff
@@ -22,7 +22,7 @@
 __summary__ = 'Automatically renders coverage reports from Chanjo ouput.'
 __uri__ = 'https://github.com/robinandeer/chanjo-report'
 
-__version__ = get_distribution(__title__).version
+__version__ = version(__title__)
```
Comment (northwestwitch) [inline chanjo_report/__init__.py:25]: This was easy

Comment (dnil) [inline chanjo_report/__init__.py:25]: Yep, no weird testing imports here I guess! 😁

**Comment by northwestwitch** [comment]
Mm I think there is something wrong in the code. I get internal error 500 on the browser

File: requirements.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 cairocffi
-chanjo>=4.7
+chanjo>=4.7.1
```
Comment (dnil) [inline requirements.txt:2]: Ahem, 4.7.2 coming up - we never released it.. 😊

Comment (northwestwitch) [reply requirements.txt:2]: Right, I assumed that was the latest. That's why i still got the pkg error! 😆

File: chanjo_report/__init__.py [hunk 3]
```diff
@@ -8,7 +8,7 @@
 :copyright: (c) 2014 by Robin Andeer
 :licence: MIT, see LICENCE for more details
 """
-from pkg_resources import get_distribution
+from importlib.metadata import version
```
[the remark this answers is not in the document]
Comment (northwestwitch) [reply chanjo_report/__init__.py:11]: I think so, I wouldn't keep supporting super old versions

**Comment by northwestwitch** [comment]
> We will still need a "pip install setuptools" or similar for this to build. If we have a separate "pip install requirements.txt" somewhere it could go in there as well, but the pare requirements from setup I think will not fly on its own.

I don't think the repo is meant to be running on its own? We install it on scout via pip and it is meant as part of chanjo. running with a command like this:

`chanjo --config demodata/chanjo.yaml report --render html`

Did I misunderstand your comment?

**Comment by northwestwitch** [comment]
I'll test it using docker and the docker-compose now

**Comment by dnil** [comment]
Precisely that - it looks like we would need to fix the Dockerfile and/or requirements for that to fly!

File: docker-compose.yml [hunk 4]
```diff
@@ -13,10 +13,12 @@ services:
...
-        test: "mysql --user=chanjoUser --password=chanjoPassword --execute \"SHOW DATABASES;\""
-        timeout: 10s
-        retries: 20
+      healthcheck:
+          test: [ "CMD", "healthcheck.sh", "--connect", "--innodb_initialized" ]
+          start_period: 10s
+          interval: 10s
+          timeout: 5s
+          retries: 3
       networks:
```
Comment (northwestwitch) [inline docker-compose.yml:22]: After this fix:

image

Comment (northwestwitch) [reply docker-compose.yml:22]: I just copied the official healthcheck step from here: https://mariadb.com/kb/en/using-healthcheck-sh/

**Comment by northwestwitch** [comment]
I think this is fine to merge now. The docker-compose file does not start a demo with data present in the database and the report page ready on the browser, but it never did. We could fix it to work like that but it's beyond the purpose of this PR I think.

To test the repo, just follow the instructions present on README:
- `make build`
- `make setup`
- `make report`

**Pull request merged**